driver-workload-executor/src/main/com/mongodb/workload/WorkloadExecutor.java

```diff
@@ -16,110 +16,152 @@
 
 package com.mongodb.workload;
 
-import com.mongodb.client.JsonPoweredCrudTestHelper;
+import com.mongodb.MongoClientSettings;
 import com.mongodb.client.MongoClient;
 import com.mongodb.client.MongoClients;
-import com.mongodb.client.MongoCollection;
-import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.unified.Entities;
+import com.mongodb.client.unified.UnifiedTest;
+import com.mongodb.diagnostics.logging.Logger;
+import com.mongodb.diagnostics.logging.Loggers;
 import org.bson.BsonArray;
 import org.bson.BsonDocument;
-import org.bson.BsonInt32;
-import org.bson.BsonValue;
+import org.bson.BsonInt64;
+import org.bson.codecs.BsonDocumentCodec;
+import org.bson.codecs.DecoderContext;
+import org.bson.json.JsonReader;
 import org.bson.json.JsonWriterSettings;
 
-import java.io.BufferedWriter;
-import java.io.File;
-import java.io.FileOutputStream;
+import java.io.FileReader;
 import java.io.IOException;
-import java.io.OutputStreamWriter;
-import java.util.concurrent.atomic.AtomicInteger;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
 
-/**
- * A workload executor for the driver.  It will execute the provided workload in a loop, until the process is terminated.
- *
- * The following system properties are supported:
- *
- *  org.mongodb.test.uri: the connection string required to connect to the cluster under test
- *  org.mongodb.workload.spec: a JSON representation of the the workload
- *  org.mongodb.workload.output.directory: the output directory in which to write the results.json
- */
 public class WorkloadExecutor {
-    private static volatile boolean done;
-    private static final AtomicInteger numSuccesses = new AtomicInteger();
-    private static final AtomicInteger numFailures = new AtomicInteger();
-    private static final AtomicInteger numErrors = new AtomicInteger();
-
-    /**
-     * The main entry point
-     *
-     * @param args no arguments are expected
-     */
-    public static void main(String[] args) {
-        String outputDirectory =
-                System.getProperty("org.mongodb.workload.output.directory", System.getProperty("user.dir"));
-        
+    private static final Logger LOGGER = Loggers.getLogger("workload-executor");
+    private static final CountDownLatch terminationLatch = new CountDownLatch(1);
+    private static volatile boolean terminateLoop;
+
+    public static void main(String[] args) throws IOException {
+        if (args.length != 2) {
+            System.out.println("Usage: AstrolabeTestRunner <path to workload spec JSON file> <path to results directory>");
```

Comment:
@rozza I left this on as a `println` as it seemed weird to use a logger to report incorrect usage.

Reply:
Ack